### PR TITLE
Jmatthewc11 windows version, fix run bug

### PR DESCRIPTION
**Pull request by jdelbick**
## Description
- fixes build on windows: replacing original PR https://github.com/adobe/asset-compute-devtool/pull/13
- fixes bug introduced with `start-prod` and `start-dev` in PR https://github.com/adobe/asset-compute-devtool/pull/6

Fixes # https://jira.corp.adobe.com/browse/NUI-630

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -26,7 +26,12 @@ app.use(express.json());
 app.use(express.urlencoded({ extended: false }));
 
 if (process.env.ASSET_COMPUTE_DEV_TOOL_ENV !== 'development') {
-    app.use(express.static(path.join(__dirname, 'client-build')));
+    if (process.platform === "darwin" || process.platform === "linux") {
+        app.use(express.static(path.join(__dirname, 'client-build')));
+    }
+    else if (process.platform === "win32") {
+        app.use(express.static(path.join(__dirname, '\\client-build')));
```
Comment (tmathern) [inline server/app.js:33]: Join is suppoed to take platforms into account when joining: (from Nodejs doc): 
`The path.join() method joins all given path segments together using the platform-specific separator as a delimiter, then normalizes the resulting path.`.

Please add a unit test for that behavior and use Travis' build capabilities for Linux and WIndows (and log the resulting joined paths). I'd like to see if platform filtering is really needed here (as long as we can avoid it, we should).

File: server/index.js [hunk 2]
```diff
@@ -110,3 +110,4 @@ async function onError(error) {
 }
 
 module.exports = { start };
+require('make-runnable');
```
Comment (tmathern) [inline server/index.js:113]: What is that? Why do we need it?

Comment (jdelbick) [reply server/index.js:113]: I added this library: https://www.npmjs.com/package/make-runnable so we can run this as a script in the command line: 
```
> node index start
```

**Comment by tmathern** [comment]
@jdelbick Regarding the test failures: Maybe the used `assert` for React is not the standard node `assert`, which may cause the failures.

**Comment by jdelbick** [comment]
@tmathern no it was just a silly typo: https://github.com/adobe/asset-compute-devtool/pull/15/commits/e12e09272d824be53f0ce4271701151e571de9f3#diff-7bb201eda24e28f3bdc97b35cc03d647R20

```
const assert = require('console');
```

**Comment by jmatthewc11** [comment]
I've almost got the **npm run build** fix for Windows, just a one line change

**Comment by tmathern** [comment]
@jdelbick @jmatthewc11 Please also finish the PR description. Thanks!

**Comment by jmatthewc11** [comment]
@delbick I tested the Windows fix for **npm run build** and it works, going to make a commit with the change.  I also tested the commit before the assertion fix, displaying in browser failed and I think it was because of the path_join() stuff

**Comment by jdelbick** [comment]
@jmatthewc11 great! thanks so much for fixing. After you push your commit, try it again with the assert fixes to make sure everything works as it should. 

If displaying in a browser still does not work, we are not done yet

**Pull request merged**